server/service/src/sync/api/core.rs

```diff
@@ -238,6 +242,30 @@ async fn response_or_err(
     Err(SyncApiErrorVariantV5::from_response_and_status(response.status(), response).await)
 }
 
+// OMS Central does not yet do auth validation for site credentials
+// So we call Legacy central server for this
+// (Use sync API for simplest auth)
+pub async fn validate_site_auth(
+    ctx: &ServiceContext,
+    sync_v5_settings: &SyncApiSettings,
+) -> Result<SiteInfoV5, CentralApiError> {
+    // We need to ignore the OG server URL provided by the remote and ensure we use the one that the OMS central server is expecting
+    let kv_repo = KeyValueStoreRepository::new(&ctx.connection);
+    let kv_url = kv_repo.get_string(KeyType::SettingsSyncUrl)?.unwrap();
```

Comment:
This unwrap is probably illegal yes?

Reply:
I think so but I prefer to leave a comment with justification when using an unwrap...

Reply:
With renewed energy I have just handled it properly by returning an error. No panics!